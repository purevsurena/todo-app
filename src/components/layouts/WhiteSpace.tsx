import React from "react";
import { WhiteSpaceProps } from "@/types/layout";
import { View } from "react-native";

const WhiteSpace: React.FC<WhiteSpaceProps> = ({ size, style }) => {
  return (
    <View
      testID="white-space"
      style={[{ height: parseInt(size, 10) }, style]}
    />
  );
};

export default WhiteSpace;
