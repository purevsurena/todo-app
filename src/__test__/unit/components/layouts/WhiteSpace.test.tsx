import React from "react";
import { render } from "@testing-library/react-native";
import WhiteSpace from "@/components/layouts/WhiteSpace";

describe("WhiteSpace Component", () => {
  it("renders correctly with given size", () => {
    const { getByTestId } = render(<WhiteSpace size="24" />);
    const whiteSpace = getByTestId("white-space");
    expect(whiteSpace.props.style).toContainEqual({ height: 24 });
  });

  it("applies additional styles correctly", () => {
    const additionalStyle = { backgroundColor: "red" };
    const { getByTestId } = render(
      <WhiteSpace size="24" style={additionalStyle} />,
    );
    const whiteSpace = getByTestId("white-space");
    expect(whiteSpace.props.style).toContainEqual({ height: 24 });
    expect(whiteSpace.props.style).toContainEqual(additionalStyle);
  });

  it("matches snapshot", () => {
    const { toJSON } = render(<WhiteSpace size="12" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
