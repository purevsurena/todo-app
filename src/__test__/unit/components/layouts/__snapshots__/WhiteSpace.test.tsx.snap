// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhiteSpace Component matches snapshot 1`] = `
<View
  style={
    [
      {
        "height": 12,
      },
      undefined,
    ]
  }
  testID="white-space"
/>
`;
